src/Data/Nat/Rough.agda

```diff
@@ -0,0 +1,62 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Natural numbers whose prime factors are all above a bound
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Nat.Rough where
+
+open import Data.Nat.Base using (ℕ; suc; _≤_; _<_; z≤n; s≤s; _+_)
+open import Data.Nat.Divisibility using (_∣_; _∤_; ∣-trans; ∣1⇒≡1)
+open import Data.Nat.Induction using (<-rec; <-Rec)
+open import Data.Nat.Primality using (Prime; composite?)
+open import Data.Nat.Properties using (_≟_; <-trans; ≤∧≢⇒<)
+open import Data.Product.Base using (_,_)
+open import Function.Base using (_∘_; flip)
+open import Relation.Nullary.Decidable.Core using (yes; no)
+open import Relation.Binary.PropositionalEquality.Core using (refl)
+
+-- A number is k-rough if all its prime factors are greater than or equal to k
+_Rough_ : ℕ → ℕ → Set
+k Rough n = ∀ {d} → 1 < d → d < k → d ∤ n
+
+-- any number is 2-rough because all primes are greater than or equal to 2
+2-rough-n : ∀ n → 2 Rough n
+2-rough-n n {1} (s≤s ()) 1<2
+2-rough-n n {suc (suc d)} 1<d (s≤s (s≤s ()))
+
+-- if a number is k-rough, and it's not a multiple of k, then it's (k+1)-rough
+extend-∤ : ∀ {k n} → k Rough n → k ∤ n → suc k Rough n
+extend-∤ {k = k} k-rough-n k∤n {suc d} 1<d (s≤s d<k) with suc d ≟ k
+... | yes refl = k∤n
+... | no  d≢k  = k-rough-n 1<d (≤∧≢⇒< d<k d≢k)
+
+-- 1 is always rough
+b-rough-1 : ∀ k → k Rough 1
+b-rough-1 k {d} 1<d d<k d∣1 with ∣1⇒≡1 d∣1
+b-rough-1 k {.1} (s≤s ()) d<k d∣1 | refl
+
+-- if a number is k-rough, then so are all of its divisors
+reduce-∣ : ∀ {k m n} → k Rough m → n ∣ m → k Rough n
+reduce-∣ k-rough-n n∣m d<k d-prime d∣n = k-rough-n d<k d-prime (∣-trans d∣n n∣m)
+
+-- if a number is k-rough, then no number 2 ≤ d < k will divide it, whether prime or not
+rough⇒∤ : ∀ {d k n} → k Rough n → 2 ≤ d → d < k → d ∤ n
+rough⇒∤ {d} {k} {n} k-rough-n = <-rec (λ d′ → 2 ≤ d′ → d′ < k → d′ ∤ n) (rough⇒∤Rec k-rough-n) d
+  where
+    rough⇒∤Rec : ∀ {n k} → k Rough n → ∀ d → <-Rec (λ d′ → 2 ≤ d′ → d′ < k → d′ ∤ n) d → 2 ≤ d → d < k → d ∤ n
+    rough⇒∤Rec {n} {k} k-rough-n (suc (suc d)) rec (s≤s (s≤s z≤n)) d<k with composite? (2 + d)
+    ... | yes (d′ , 2≤d′ , d′<d , d′∣d) = rec d′ d′<d 2≤d′ (<-trans d′<d d<k) ∘ ∣-trans d′∣d
+    ... | no ¬d-composite = k-rough-n {2 + d} (s≤s (s≤s z≤n)) d<k
```

Comment:
Shouldn't this be a trivial consequence of `1 < 2` instead of a proof by strong induction?